java/016_3Sum_Closest.java

```diff
@@ -0,0 +1,66 @@
+import java.util.Arrays;
+
+//16. 3Sum Closest
+/**
+* Given an array S of n integers, find three integers in S such that the sum is closest to a given number, target. Return the sum of the three integers. You may assume that each input would have exactly one solution.
+* 
+*     For example, given array S = {-1 2 1 -4}, and target = 1.
+* 
+*     The sum that is closest to the target is 2. (-1 + 2 + 1 = 2).
+* 
+* Tags: Array, Two Pointers
+* 
+* Similar Problems: (M) 3Sum, (M) 3Sum Smaller
+* 
+* Author: Xinyue Zhang
+*/
+
+public class Solution {
+    public int threeSumClosest(int[] nums, int target) {
+        Arrays.sort(nums);
+        int n = nums.length;
+        if (n < 3) return 0;
+        
+        //if maximum smaller than target, then return
+        int res = nums[n - 1] + nums[n - 2] + nums[n - 3];
+        if (res < target) return res;
```

Comment:
When `res` is equal to `target`, it should also return `res`. It will save a lot of time.